### PR TITLE
driver/powerdriver: Properly reconstruct proxied URL

### PR DESCRIPTION
**Pull request by edersondisouza**
Some PDUs URL include username/password and query attributes, and they need to be included in the proxied URL as well.

**Description**
Include URL based username, password and query attributes on proxied URL (those used when `labgrid-exporter` is started with `-i` option, making all connections go via SHH tunnerl).

### CODE REVIEW

File: labgrid/driver/powerdriver.py [hunk 1]
```diff
@@ -203,7 +207,8 @@ def set_proxy_from_url(self):
         )
 
         # construct proxied host parameter
-        self._host = f'{url.scheme}://{self._host}:{self._port}{url.path}'
+        query = f'?{url.query}' if url.query else ''
```
[the remark this answers is not in the document]
Comment (edersondisouza) [reply labgrid/driver/powerdriver.py:206]: Got it!

**Comment by edersondisouza** [comment]
v2:
 - Improve consistency and locality of change.

**Review by Bastian-Krause — APPROVED** [review]
Thanks. I've also added a test case for the basic auth credentials in URL case.

**Review by Emantor — APPROVED** [review]

**Comment by jluebbe** [comment]
Perhaps it's a bit late... @edersondisouza Do you remember why you didn't use `.username` and `.password` instead of `.netloc`?

**Comment by edersondisouza** [comment]
> Perhaps it's a bit late... @edersondisouza Do you remember why you didn't use `.username` and `.password` instead of `.netloc`?

I think that ends up being simpler, maybe? So I don't have to care about adding the `:` when there's a password, I just reuse the whole `username[:<password>]` block.

**Pull request merged**